doc/specs/#3061 - Marketing Version.md

```diff
@@ -0,0 +1,124 @@
+---
+author: Kaleb Luedtke (Trenly) trenlymc@gmail.com
+created on: 2023-03-10
+last updated: 2023-03-10
+issue id: 3061
+---
+
+# Marketing Version
+
+"For [#3061](https://github.com/microsoft/winget-cli/issues/3061)"
+
+## Abstract
+
+There are times when the version which is marketed to consumers on publishers' websites differs from what is written to the Apps and Features Entries data. Additionally, there are times when the same version is marketed for several different versions of the same package. The Windows Package Manager should be able to show an arbitrary version as determined by the publisher.
+
+## Inspiration
+
+The current design using `PackageVersion` for the displayed version and how it differs from `DisplayVersion` can be confusing.
+
+## Solution Design
+
+The Windows Package Manager manifest schema should have a `MarketingVersion` key added at the root level. For clarity, it should be placed immediately following the `PackageVersion` key.
+
+When the Windows Package Manager needs to display the version of a package, the value from the `MarketingVersion` key should be used. If the key is not present or has no value, the value of `PackageVersion` should be displayed instead, with the exception of `winget show <query> --versions` and `winget search <query> --versions` which should show the values as key pairs of `PackageVersion (MarketingVersion)`.
+
+When the `--version` parameter is used, matches with `PackageVersion` should be considered first and matches with `MarketingVersion` should be considered second. If there are multiple package versions of the same identifier matched using `MarketingVersion`, the highest `PackageVersion` shall be selected.
+
+When `MarketingVersion` is present, `PackageVersion` should be validated to be a semantic version.
+
+For the purposes of version comparison, mapping installed packages to versions, and all other behaviors, the current implementation using `PackageVersion` should be retained.
+
+## UI/UX Design
+
+Example of a manifest with a `MarketingVersion` specified:
+```yaml
+PackageIdentifier: Example.Package
+PackageVersion: 1.2.5.1
+MarketingVersion: 1.2.5
```

Comment:
Do you think MarketingVersion should exist in default locale manifest only Or it could be different for different locales?

Reply:
That's an interesting point. I would presume that it would be the same across locales, since there is usually a single version referenced on websites / documentation. I suppose it is possible that different locales could use different versions, but I would be a very small number of packages. Perhaps it would be better in the version manifest then, instead of the locale?

Reply:
Version manifest is currently only used for validation and ignored during processing (merging in the service). If we want only one marketing version, we can limit it to just the default locale.